boards/shields/waveshare_at24/waveshare_at24.overlay

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2020 Stefano Manni
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&i2c1 {
```

Comment:
Depending on boards, arduino i2c might not be i2c '1' instance but '0', '2' or even 'a'.
Shield concept relies on the use of standard connectors (arduino would be the most common one), so we can use abstractions in dts files.
As a consequence, if shield is using standard arduino I2C, please use `&arduino_i2c` here.
If it isn't, then you need to propose a board specific overlay:
- leave this overlay file empty
- add a board specific overlay pointing to i2c1 under `boards/shields/waveshare_at24/boards/tatgerted_board.overlay`

Reply:
thanks for pointing it out, done in 21dff0f

Reply:
Sorry, I might have not been clear enough.
`boards/shields/waveshare_at24/boards/waveshare_at24.overlay` is not a proper file name.
This should be for instance `boards/shields/waveshare_at24/boards/nucleo_f401re.overlay` if you're targeting nucleo_f401re.
But first 2 questions:
-on which board have you tested this shield ?
-are you sure this board is not compatible with i2c arduino device generic pins ?

Reply:
You sure the path you pasted is correct?

boards/shields/waveshare_at24/boards/waveshare_at24.overlay
it does not esists. Probably you meant:
boards/shields/waveshare_at24/boards/waveshare_open103z.overlay

I've tested it on a waveshare_open103 that unfortunately does not have an arduino compatible connector.

Reply:
Ok, looking closer to the shield it seems it doesn't feature a arduino connector at all (it seems I made it up myslef).
Hence, this is component is not exactly a shield.
While the zephyr shield definition doesn't mention a shield has to feature a standard connector, I'd prefer not seeing a bunch of ad hoc components spreading in.
I'm giving a go for this one, but will watch evolution.

Reply:
If a shield is something with a standard connector, no that's not a shield I agree with you.
Since I'd like to add more 'shields' from waveshare like this featuring different components (spi, eth, tft, etc.) whats the proper way to add them? Please advise.